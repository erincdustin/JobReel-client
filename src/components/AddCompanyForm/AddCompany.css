@media screen and (min-width: 960px) {
  .AddCompanyForm {
    border: 1px black solid;
    min-width: 300px;
    margin: 0 auto;
    padding: 5%;
    line-height: 150%;
    border-radius: 10px;
    text-align: center;
    margin: auto;
    background: #3a3846;
    color: white;
    grid-column-start: 4;
    grid-column-end: 7;
    box-shadow:  0.5em 0.5em 0.2em rgb(48, 39, 39);
  }
  
  #state-input {
    margin: 0 auto;
    border-radius: 10px;
    width: 15em;
    height: 2em;
  }
}

@media screen and (max-width: 960px) {
  .AddCompanyForm {
    border: 1px black solid;
    min-width: 300px;
    margin: 0 auto;
    padding: 5%;
    line-height: 150%;
    border-radius: 10px;
    text-align: center;
    margin: auto;
    background: #3b3a50;
    color: #eeebd9;
    grid-column-start: 1;
    grid-column-end: 9;
    box-shadow:  0.5em 0.5em 0.2em rgb(48, 39, 39);
  }
  
  #state-input {
    margin: 0 auto;
    border-radius: 10px;
    width: 15em;
    height: 2em;
  }
}

