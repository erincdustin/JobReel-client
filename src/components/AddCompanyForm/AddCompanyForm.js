import React from 'react';
import { Input, Label } from '../../components/Form/Form';
import Button from '../../components/Button/Button';
import JobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';
import './AddCompany.css';

class AddCompanyForm extends React.Component {

  static contextType = JobReelContext

  state = {
    error: null
  }

  handleSubmit = (e) => {
    e.preventDefault();
    e.persist();
    this.setState({ error: null });
    const company_name = e.target.company.value;
    const city = e.target.city.value;
    const state = e.target.state.value;
    const industry = e.target.industry.value;
    const website = e.target.website.value;
    const description = e.target.desc.value;
    const contact = e.target.contact.value;
    const userInput = { user_id: this.context.user.id, company_name, city, state, industry, website, description, contact };
    jobReelApiService.submitCompany(userInput)
      .then(res => {
        e.target.company.value = '';
        e.target.city.value = '';
        e.target.state.value = '';
        e.target.industry.value = '';
        e.target.website.value = '';
        e.target.desc.value = '';
        e.target.contact.value = '';
        this.context.setCompanies([...this.context.companies, res]);
        this.context.setManualCompanyAdd(false)
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  renderStateOptions = () => {
    const states = [
      ['Arizona', 'AZ'],
      ['Alabama', 'AL'],
      ['Alaska', 'AK'],
      ['Arkansas', 'AR'],
      ['California', 'CA'],
      ['Colorado', 'CO'],
      ['Connecticut', 'CT'],
      ['Delaware', 'DE'],
      ['Florida', 'FL'],
      ['Georgia', 'GA'],
      ['Hawaii', 'HI'],
      ['Idaho', 'ID'],
      ['Illinois', 'IL'],
      ['Indiana', 'IN'],
      ['Iowa', 'IA'],
      ['Kansas', 'KS'],
      ['Kentucky', 'KY'],
      ['Louisiana', 'LA'],
      ['Maine', 'ME'],
      ['Maryland', 'MD'],
      ['Massachusetts', 'MA'],
      ['Michigan', 'MI'],
      ['Minnesota', 'MN'],
      ['Mississippi', 'MS'],
      ['Missouri', 'MO'],
      ['Montana', 'MT'],
      ['Nebraska', 'NE'],
      ['Nevada', 'NV'],
      ['New Hampshire', 'NH'],
      ['New Jersey', 'NJ'],
      ['New Mexico', 'NM'],
      ['New York', 'NY'],
      ['North Carolina', 'NC'],
      ['North Dakota', 'ND'],
      ['Ohio', 'OH'],
      ['Oklahoma', 'OK'],
      ['Oregon', 'OR'],
      ['Pennsylvania', 'PA'],
      ['Rhode Island', 'RI'],
      ['South Carolina', 'SC'],
      ['South Dakota', 'SD'],
      ['Tennessee', 'TN'],
      ['Texas', 'TX'],
      ['Utah', 'UT'],
      ['Vermont', 'VT'],
      ['Virginia', 'VA'],
      ['Washington', 'WA'],
      ['West Virginia', 'WV'],
      ['Wisconsin', 'WI'],
      ['Wyoming', 'WY'],
    ];
    return states.map(state => {
      return <option key={state[1]} value={state[1]}>{state[0]}</option>
    })
  }

  render() {
    const { error } = this.state;
    return (
      <div className="AddCompanyForm">
        <div role='alert'>
          {error && <p>{error}</p>}
        </div>
        <form onSubmit={this.handleSubmit}>
          <div>
            <Label htmlFor='company-name-input'>
              Company
            </Label>
            <br/>
            <Input
              id='company-name-input'
              name='company'
            />
          </div>
          <div>
            <Label htmlFor='city-input'>
              City
            </Label>
            <br/>
            <Input
              id='city-input'
              name='city'
            />
          </div>
          <div>
            <Label htmlFor='state-input'>
              State
            </Label>
            <br/>
            <select name="state" id="state-input">
              {this.renderStateOptions()}
            </select>
          </div>
          <div>
            <Label htmlFor='industry-input'>
              Industry
            </Label>
            <br/>
            <Input
              id='industry-input'
              name='industry'
            />
          </div>
          <div>
            <Label htmlFor='website-input'>
              Website
            </Label>
            <br/>
            <Input
              id='website-input'
              name='website'
            />
          </div>
          <div>
            <Label htmlFor='desc-input'>
              Description
            </Label>
            <br/>
            <Input
              id='desc-input'
              name='desc'
            />
          </div>
          <div>
            <Label htmlFor='contact-input'>
              Contact
            </Label>
            <br/>
            <Input
              id='contact-input'
              name='contact'
            />
          </div>
          <Button onClick={() => this.context.setManualCompanyAdd(false)} type="button">Back</Button>
          <Button type="submit">Submit</Button>
        </form>
      </div>
    )
  }
}

export default AddCompanyForm;