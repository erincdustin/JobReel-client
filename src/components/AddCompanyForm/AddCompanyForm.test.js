import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import AddCompanyForm from './AddCompanyForm'

describe('AddCompanyForm smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddCompanyForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})

describe('AddCompanyForm snapshot test', () => {
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<AddCompanyForm />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})