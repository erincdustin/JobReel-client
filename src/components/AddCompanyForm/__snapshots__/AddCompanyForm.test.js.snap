// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCompanyForm snapshot test renders the UI as expected 1`] = `
<div
  className="AddCompanyForm"
>
  <div
    role="alert"
  />
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        className="Label"
        htmlFor="company-name-input"
      >
        Company
      </label>
      <br />
      <input
        className="Input"
        id="company-name-input"
        name="company"
        type="text"
      />
    </div>
    <div>
      <label
        className="Label"
        htmlFor="city-input"
      >
        City
      </label>
      <br />
      <input
        className="Input"
        id="city-input"
        name="city"
        type="text"
      />
    </div>
    <div>
      <label
        className="Label"
        htmlFor="state-input"
      >
        State
      </label>
      <br />
      <select
        id="state-input"
        name="state"
      >
        <option
          value="AZ"
        >
          Arizona
        </option>
        <option
          value="AL"
        >
          Alabama
        </option>
        <option
          value="AK"
        >
          Alaska
        </option>
        <option
          value="AR"
        >
          Arkansas
        </option>
        <option
          value="CA"
        >
          California
        </option>
        <option
          value="CO"
        >
          Colorado
        </option>
        <option
          value="CT"
        >
          Connecticut
        </option>
        <option
          value="DE"
        >
          Delaware
        </option>
        <option
          value="FL"
        >
          Florida
        </option>
        <option
          value="GA"
        >
          Georgia
        </option>
        <option
          value="HI"
        >
          Hawaii
        </option>
        <option
          value="ID"
        >
          Idaho
        </option>
        <option
          value="IL"
        >
          Illinois
        </option>
        <option
          value="IN"
        >
          Indiana
        </option>
        <option
          value="IA"
        >
          Iowa
        </option>
        <option
          value="KS"
        >
          Kansas
        </option>
        <option
          value="KY"
        >
          Kentucky
        </option>
        <option
          value="LA"
        >
          Louisiana
        </option>
        <option
          value="ME"
        >
          Maine
        </option>
        <option
          value="MD"
        >
          Maryland
        </option>
        <option
          value="MA"
        >
          Massachusetts
        </option>
        <option
          value="MI"
        >
          Michigan
        </option>
        <option
          value="MN"
        >
          Minnesota
        </option>
        <option
          value="MS"
        >
          Mississippi
        </option>
        <option
          value="MO"
        >
          Missouri
        </option>
        <option
          value="MT"
        >
          Montana
        </option>
        <option
          value="NE"
        >
          Nebraska
        </option>
        <option
          value="NV"
        >
          Nevada
        </option>
        <option
          value="NH"
        >
          New Hampshire
        </option>
        <option
          value="NJ"
        >
          New Jersey
        </option>
        <option
          value="NM"
        >
          New Mexico
        </option>
        <option
          value="NY"
        >
          New York
        </option>
        <option
          value="NC"
        >
          North Carolina
        </option>
        <option
          value="ND"
        >
          North Dakota
        </option>
        <option
          value="OH"
        >
          Ohio
        </option>
        <option
          value="OK"
        >
          Oklahoma
        </option>
        <option
          value="OR"
        >
          Oregon
        </option>
        <option
          value="PA"
        >
          Pennsylvania
        </option>
        <option
          value="RI"
        >
          Rhode Island
        </option>
        <option
          value="SC"
        >
          South Carolina
        </option>
        <option
          value="SD"
        >
          South Dakota
        </option>
        <option
          value="TN"
        >
          Tennessee
        </option>
        <option
          value="TX"
        >
          Texas
        </option>
        <option
          value="UT"
        >
          Utah
        </option>
        <option
          value="VT"
        >
          Vermont
        </option>
        <option
          value="VA"
        >
          Virginia
        </option>
        <option
          value="WA"
        >
          Washington
        </option>
        <option
          value="WV"
        >
          West Virginia
        </option>
        <option
          value="WI"
        >
          Wisconsin
        </option>
        <option
          value="WY"
        >
          Wyoming
        </option>
      </select>
    </div>
    <div>
      <label
        className="Label"
        htmlFor="industry-input"
      >
        Industry
      </label>
      <br />
      <input
        className="Input"
        id="industry-input"
        name="industry"
        type="text"
      />
    </div>
    <div>
      <label
        className="Label"
        htmlFor="website-input"
      >
        Website
      </label>
      <br />
      <input
        className="Input"
        id="website-input"
        name="website"
        type="text"
      />
    </div>
    <div>
      <label
        className="Label"
        htmlFor="desc-input"
      >
        Description
      </label>
      <br />
      <input
        className="Input"
        id="desc-input"
        name="desc"
        type="text"
      />
    </div>
    <div>
      <label
        className="Label"
        htmlFor="contact-input"
      >
        Contact
      </label>
      <br />
      <input
        className="Input"
        id="contact-input"
        name="contact"
        type="text"
      />
    </div>
    <button
      className="Button"
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
    <button
      className="Button"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
